profiler/upload.go

```diff
@@ -88,6 +88,9 @@ func (p *profiler) doRequest(bat batch) error {
 		cancel()
 	}()
 	req, err := http.NewRequestWithContext(ctx, "POST", p.cfg.targetURL, body)
+	if err != nil {
```

Comment:
This is an unrelated change, but I noticed there was no error check while I was investigating whether stopping the upload was actually the inconsistent part leading to test flakes.

Reply:
👍 Seems like we missed this small regression in https://github.com/DataDog/dd-trace-go/pull/1239 (cc @ajgajg1134). But I think this method will only return an error if we pass in invalid arguments, so it's probably not been having any impact.